backend.py

```diff
@@ -440,6 +441,19 @@ def performInstallation(answers, ui_package, interactive):
                                "latest dedicated ISO." %
                                (answers['linstor-version'], ', '.join(available_linstor_versions)))
 
+    # if upgrading a host with xapi, check we can upgrade it first
+    if answers['install-type'] == INSTALL_TYPE_REINSTALL and answers['xapi-version']:
+        available_xapi_versions = repository.listPackagesFromRepos(
+            main_repositories, 'xapi-core', '%{evr}', latest_only=True)
+        if not available_xapi_versions:
+            raise RuntimeError("No XAPI package found in package source")
+        assert len(available_xapi_versions) == 1
+        if rpm.labelCompare(('pkg', 'arch', available_xapi_versions[0]), ('pkg', 'arch', answers['xapi-version'])) < 0:
+            raise RuntimeError("Cannot upgrade the host: the XAPI version on the host (%s) "
```

Comment:
Didn't we just say that it may be better to make it a warning screen rather than a fatal error?